src/ArcxAnalyticsSdk.ts

```diff
@@ -75,6 +95,80 @@ export class ArcxAnalyticsSdk {
     )
   }
 
+  private async _onAccountsChanged(accounts: string[]) {
+    if (!window.ethereum) {
+      throw new Error('ArcxAnalyticsSdk::_onAccountsChanged: No ethereum provider found')
+    }
+
+    if (accounts.length > 0) {
+      if (accounts[0] === this.previousConnectedAccount) {
+        // We have already reported this account
+        return
+      } else {
+        this.previousConnectedAccount = accounts[0]
+      }
+
+      const chainIdHex = await window.ethereum.request<string>({ method: 'eth_chainId' })
+      // Because we're connected, the chainId cannot be null
+      if (!chainIdHex) {
+        throw new Error('ArcxAnalyticsSdk::_onAccountsChanged: chainIdHex is:' + chainIdHex)
+      }
+      this.previousChainId = parseInt(chainIdHex, 16).toString()
+
+      this.connectWallet({ chain: this.previousChainId, account: accounts[0] })
```

Comment:
do you think it is important to difirinciate automatic events and manual?

Reply:
`this.previousChainId` it is `actualChainId` for now

Reply:
renamed variables